backend/integrations/oauth/google.go

```diff
@@ -0,0 +1,51 @@
+package oauth
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/GenerateNU/sac/backend/config"
+)
+
+type GoogleOAuthClient struct {
+	OAuthConfig config.OAuthSettings
+}
+
+func (client *GoogleOAuthClient) AuthorizeURL(state string) string {
+	return (client.OAuthConfig.BaseURL + "/auth?" +
```

Comment:
here is an example of using the stdlib's url type to build -> https://noahkreiger.medium.com/golang-building-dynamic-urls-4e93cab86e72